ln1_f4.py

```diff
@@ -8,86 +8,188 @@
-systime_customformat = systime_customformat.strftime("%d,%m,%Y,%H,%M,%S")
-
-sysinfo_variables_names = [cpu_total, cpu_user, cpu_sys, cpu_idle, mem_total, mem_used, mem_free,
-                           mem_cached, hdd_total, hdd_used, hdd_free]
-sysinfo_database_keys = ["cpu_total", "cpu_user", "cpu_sys", "cpu_idle", "mem_total", "mem_used", "mem_free",
-                         "mem_cached", "hdd_total", "hdd_used", "hdd_free"]
+    command_data, error_data = command_data.communicate()
+    return command_data.replace(',', '.').split()
+
+
+def get_cpu_info():
+    cpu_console_data = get_sysresinfo('mpstat')
+    cpu_user = float(cpu_console_data[21])
+    cpu_sys = sum(float(x) for x in cpu_console_data[22:29])
+    cpu_total = cpu_user + cpu_sys
+    cpu_idle = float(cpu_console_data[30])
+    return {'cpu_user': cpu_user, 'cpu_sys': cpu_sys, 'cpu_total': cpu_total, 'cpu_idle': cpu_idle}
+
+
+def get_mem_info():
+    mem_console_data = get_sysresinfo('free -m')
+    mem_total, mem_used = int(mem_console_data[7]), int(mem_console_data[8])
+    mem_free, mem_cached = int(mem_console_data[9]), int(mem_console_data[12])
+    return {'mem_total': mem_total, 'mem_used': mem_used, 'mem_free': mem_free, 'mem_cached': mem_cached}
 
-###############################################################################################################
 
-if os.path.isfile("/home/lax/PycharmProjects/learn1/database_cron") == False:  # Adding new record into database
-    with open("/home/lax/PycharmProjects/learn1/database_cron", "w") as database_file:
-        sysinfo_database = {"0": {
-            "email_send_hour": 24
-        }}
-        database_file.write(json.dumps(sysinfo_database))
-
-with open("/home/lax/PycharmProjects/learn1/database_cron", "r") as database_file:
-    data = database_file.read().strip()
-sysinfo_database = json.loads(data)
-sysdb_totalrecords = len(sysinfo_database)
-sysinfo_database[str(sysdb_totalrecords)] = {
-    "record_time": systime_customformat,
-    "cpu_total": cpu_total,
-    "cpu_user": cpu_user,
-    "cpu_sys": cpu_sys,
-    "cpu_idle": cpu_idle,
-    "mem_total": mem_total,
-    "mem_used": mem_used,
-    "mem_free": mem_free,
-    "mem_cached": mem_cached,
-    "hdd_total": hdd_total,
-    "hdd_used": hdd_used,
-    "hdd_free": hdd_free
-}
-with open("/home/lax/PycharmProjects/learn1/database_cron", "w") as database_file:
-    database_file.write(json.dumps(sysinfo_database))
+def get_hdd_info():
+    hdd_console_data = get_sysresinfo('df -m --total')
+    hdd_total, hdd_used = int(hdd_console_data[50]), int(hdd_console_data[51])
+    hdd_free = int(hdd_console_data[52])
+    return {'hdd_total': hdd_total, 'hdd_used': hdd_used, 'hdd_free': hdd_free}
 
 ###############################################################################################################
-dynamic_html_table = """
+
+
+class SysinfoDatabase(object):
+    """ Current database structure
+    {'2016,05,19,01,05,05': {
```

Comment:
What if you customer ask you to update your keys for more readable and in USA format(year-day-month), for example to `2016-19-05 01:05:05`. All your database logic crash and you should rewrite all your's calculation. That smart solution (just put in a row), but not so good when you try to create flexible and good readable for other programmers service.

To prevent that use `datetime.strftime` and `datetime.strptime` methods to handle all datetime converts and add `DATETIME_TEMPLATE = ''%Y-%m-%d %H-%M-%S''` as class variable constant(or as input variable in init)